### PR TITLE
React native Application

### PR DESCRIPTION
**Pull request by JeniaSavenko**

### CODE REVIEW

File: src/actions/PostActions.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+export const ERROR_GET_POSTS = 'ERROR_GET_POSTS';
+
+export const getPosts = () => ({
+  type: GET_POSTS,
+});
+
+export const postPosts = (text, title) => ({
+  type: POST_NOTES_SUCCESS,
+  title,
+  text
```
Comment (esnts) [inline src/actions/PostActions.js:20]: `text,` use one code style

File: src/actions/PostActions.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+});
+
+export function rmPost(item) {
+  return {
```
Comment (esnts) [inline src/actions/PostActions.js:24]: `arrow function`.  use one code style

File: src/components/Block/index.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+const borderProps = separateLiteralToProps`
+	${({ corners }) => corners && `border-radius: ${corners};`}
+`;
+
+const transformProps = separateLiteralToProps`
+	${({ scale }) => scale && `transform: scale(${(scale)});`}
+	${({ rotate }) => rotate && `transform: rotate(${(rotate)}deg);`}
+`;
+const isExist = value => value !== undefined;
```
Comment (esnts) [inline src/components/Block/index.js:63]: ????????? pos

File: src/components/Block/index.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  borderProps,
+  transformProps,
+  paddingProps,
+  marginProps,
+  positionProps,
+  superProps
+);
+
+const Block = applyProps(styled.View, blockProps);
+
```
Comment (esnts) [inline src/components/Block/index.js:116]: unnecessary lines

File: src/components/ListItem.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+          {title}
+        </Title>
+        <Body>
+          {text}
+        </Body>
+      </Block>
+    </Block>
+  </TouchableOpacity>
+);
+
```
Comment (esnts) [inline src/components/ListItem.js:30]: unnecessary lines

File: src/components/Navigator.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import {
+  createStackNavigator,
+  createAppContainer,
```
Comment (esnts) [inline src/components/Navigator.js:4]: unnecessary line breaker

File: src/components/Navigator.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+} from 'react-navigation';
+import HomeScreen from '../screen/HomeScreen';
+import PostPreviewScreen from '../screen/PostPreviewScreen';
+import AddPostScreen from '../screen/AddPostScreen';
+
+
+const Navigator = createStackNavigator({
+  Home: {
+    screen: HomeScreen,
+    navigationOptions: {
```
Comment (esnts) [inline src/components/Navigator.js:14]: is it necessary?  `navigationOptions: {`

File: src/components/Navigator.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+      title: 'Mega News',
+    }
+  },
+  PostScreen: { screen: PostPreviewScreen },
+  AddPost: { screen: AddPostScreen }
+
+},
+{
+  initialRouteName: 'Home',
+});
```
Comment (esnts) [inline src/components/Navigator.js:24]: unnecessary

File: src/reducers/post.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+  EDIT_MODE_POST,
+  HANDLE_TEXT_CHANGE,
+  SAVE_POST,
+  POST_NOTES_SUCCESS,
+  ERROR_GET_POSTS
+} from '../actions/PostActions';
+
+const initialState = {
+  posts: [{ _id: '1', title: 'test', text: 'text text' }],
+};
```
Comment (esnts) [inline src/reducers/post.js:15]: it likes KOCTULb

File: src/reducers/post.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+  editableField: postItem.text
+}));
+
+const savePosts = (posts, id) => posts.map(postItem => ({
+  ...postItem,
+  text: postItem.editableField,
+  editMode: false
+}));
+
+const handleTextChange = (posts, id, editableField) => posts.map(postItem => ({
```
Comment (esnts) [inline src/reducers/post.js:29]: ??????

File: src/sagas/index.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+import * as axios from 'axios';
+import {
+  CREATE_POST,
+  GET_POSTS,
+  POSTS_SUCCESS,
+  POST_NOTES_SUCCESS,
+  DELETE_NOTES_SUCCESS, SAVE_POST, ERROR_GET_POSTS,
+} from '../actions/PostActions';
+
+const url = 'http://localhost:5000/notes';
```
Comment (esnts) [inline src/sagas/index.js:13]: move to constants file, use literals

File: src/sagas/index.js [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+const combineRequests = {
```
Comment (esnts) [inline src/sagas/index.js:15]: ??????

File: src/screen/AddPostScreen.js [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+  const [title, setTitle] = useState();
+
+  const [text, setText] = useState();
+
+  return (
+    <Block>
+      <Input
+        placeholder="Title"
+        value={title}
+        onChangeText={value => setTitle(value)}
```
Comment (esnts) [inline src/screen/AddPostScreen.js:20]: `onChangeText={setTitle}`

File: src/screen/AddPostScreen.js [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+const AddPostScreen = ({ setTitleAction, navigation }) => {
+  const [title, setTitle] = useState();
+
+  const [text, setText] = useState();
+
+  return (
+    <Block>
+      <Input
+        placeholder="Title"
```
Comment (esnts) [inline src/screen/AddPostScreen.js:18]: i18n

File: src/screen/AddPostScreen.js [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+      <Input
+        placeholder="Text"
+        value={text}
+        onChangeText={value => setText(value)}
+      />
+      <Block ph="5">
+        <Button
+          title="Add"
+          onPress={() => {
+            setTitleAction(title, text) && navigation.navigate('Home');
```
Comment (esnts) [inline src/screen/AddPostScreen.js:31]:
```
 setTitleAction(title, text);
 navigation.navigate('Home');
```

Comment (esnts) [reply src/screen/AddPostScreen.js:31]: also, move it to separate function

Comment (esnts) [reply src/screen/AddPostScreen.js:31]: also, use constants for routes

File: src/screen/HomeScreen.js [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+import {
+  getPosts, rmPost, savePost, textEdit
+} from '../actions/PostActions';
+
+
+const HomeScreen = ({
+  posts,
+  getPostsAction,
+  navigation,
+}) => {
```
Comment (esnts) [inline src/screen/HomeScreen.js:18]: What is the reason to set line breakers before all arguments?

File: src/store/socket.js [hunk 17]
```diff
@@ -0,0 +1,30 @@
+import io from 'socket.io-client';
+import { POSTS_SUCCESS } from '../actions/PostActions';
+
+const token = '';
+
+const url = io('http://localhost:5000/notes');
```
Comment (esnts) [inline src/store/socket.js:6]: string duplicate

File: src/store/socket.js [hunk 18]
```diff
@@ -0,0 +1,30 @@
+import io from 'socket.io-client';
```
Comment (esnts) [inline src/store/socket.js:1]: I think, that es6 class will be a good future for this code

File: src/components/Block/index.js [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+const superProps = separateLiteralToProps`
+	${({ relative }) => relative && 'position: relative;'}
+	${({ absolute }) => absolute && 'position: absolute;'}
+	${({ centred }) => centred && 'justify-content: center; align-items: center;'}
+	${({ js }) => js && 'justify-content: flex-start;'}
+	${({ jc }) => jc && 'justify-content: center;'}
+	${({ je }) => je && 'justify-content: flex-end;'}
+	${({ jb }) => jb && 'justify-content: space-between;'}
+	${({ as }) => as && 'align-items: flex-start;'}
+	${({ ac }) => ac && 'align-items: center;'}
```
Comment (MeyerPV) [inline src/components/Block/index.js:82]: For a user who does not know the project, such a name would be a terrible headache; choose more humane names
` <Block row ac>` i mean ac,as,je & e.t.c.

File: src/screen/HomeScreen.js [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+                )}
+      >
+        <FlatList
+          data={posts}
+          renderItem={renderItem}
+          initialNumToRender={10}
+          maxToRenderPerBatch={10}
+        />
+      </ScrollView>
+      <Button title="Add Posts" onPress={() => navigation.navigate('AddPost')} />
```
Comment (MeyerPV) [inline src/screen/HomeScreen.js:60]: Why you ignore my comments, i said that you should create separate function for navigation? Avoid using arrow-functions in onPress

File: src/screen/HomeScreen.js [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+
+    posts = getPostsAction();
+  };
+
+  const renderItem = ({ item }) => (
+    <ListItem
+      title={item.title}
+      text={item.text}
+      onPress={() => {
+        navigation.navigate('PostScreen', {
```
Comment (MeyerPV) [inline src/screen/HomeScreen.js:36]: Why you ignore my comments, i said that you should create separate function for navigation? Avoid using arrow-functions in onPress

File: src/screen/PostPreviewScreen.js [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+          <Button
+            outline
+            icon={(
+              <Icon
+                name="ios-trash"
+                type="ionicon"
+                color="#fff"
+              />
+)}
+            titleStyle={{
```
Comment (MeyerPV) [inline src/screen/PostPreviewScreen.js:63]: Should be in separate file

File: src/store/socket.js [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+
+const token = '';
+
+const url = io('http://localhost:5000/notes');
+
+
+let socket;
+let store;
+
+export const configureSocket = str => {
```
Comment (MeyerPV) [inline src/store/socket.js:12]: `str `it is **bad** naming.

**Review by MeyerPV — COMMENTED** [review]
Fix my comments and stop ignore navigation

**Pull request closed without merging**